### ISSUE TITLE
Chotto Shot (Japan) - UCJB98012 extremely bad lag

### ISSUE THREAD

**Issue by benderscruffy** [hrydgard/ppsspp#10980]
v1.5.4-933-g386c9d4fc
it goes so slow it takes 15-20 mins to get to the titles screen
it keeps printing this in the log 
41:54:125 bhCameraDevi E[HLE]: HLE\HLE.cpp:545 Unimplemented HLE function sceUsbWaitState
[ppsspplog.zip](https://github.com/hrydgard/ppsspp/files/1957486/ppsspplog.zip)

**Comment by LunaMoo** [comment]
That's not "lag", this camera app is just awaiting usb state and since we don't emulate it at all we send and error which makes the app continuosly await.
 We could simply lie here and send "all ok"(aka 0) for that function(or just replace a call to it via cwcheat with "move v0,zero"(or "li v0,1"), but it wouldn't do much anyway without a camera which currently is only partially supported and only on one platform - android.

**Comment by Rafidmuhaiminwafi** [comment]
I think it can be closed. Also it can be labeled with feature request. Also it can be tagged with the future milestone.

**Comment by LunaMoo** [comment]
Trying to implement sceUsbWaitState to fix this, but I don't understand what it should be doing so I'm going to give up. It has 3 parameters which supposedly are state(usb), waitmode and timeout.

 What seems to work is returning usbState by this function, and that's basically what jpcsp is doing right now, not sure if it's worth adding a possibly wrong code just for the sake of one pointless app, so instead I'll just drop a ready to use cheat workaround here that speeds up boot:
````
_S UCJB-98012
_G Chotto Shot
_C0 QuickBoot workaround
_L 0xE00118AE 0x0003BA6C
_L 0x2003BA6C 0x00001021
_C0 QuickBoot workaround [Disable]
_L 0xE0011021 0x0003BA6C
_L 0x2003BA6C 0x0E2518AE
````
this is also wrong, but at least affects only this app.

**Comment by unknownbrackets** [comment]
It's probably similar to the umd wait?

My guess is that:
 * sceUsbWaitState waits for state, with mode choosing an exact or bit and match, and timeout optional.
 * sceUsbWaitCancel probably makes all waits on all threads cancel.

But that's just a guess, would need testing to implement.

-[Unknown]

**Comment by Panderner** [comment]
It needs camera support this is done by #8521 I've tested Invizimals

**Comment by Panderner** [comment]
It seems This game requires camera support like Invizimals and EyePet.

**Comment by LunaMoo** [comment]
It's NOT a game. It's an utility app that comes with one of the two camera's available for PSP.

 And really Panderner you should stop spamming obvious things for the sake of attention. It's an issue tracker and being captain obvious here is just annoying to people subscribed until everyone puts you on ignore.

**Comment by Panderner** [comment]
> It's NOT a game. It's an utility app that comes with one of the two camera's available for PSP.
> 
> And really Panderner you should stop spamming obvious things for the sake of attention. It's an issue tracker and being captain obvious here is just annoying to people subscribed until everyone puts you on ignore.

I don't spam, chotto shot is a editing software Which is edit photos and video clips using PSP camera. And this app is only available in Japan

**Comment by unknownbrackets** [comment]
This was probably fixed by #14149.  Sorry for forgetting to link it.

-[Unknown]

**Comment by Panderner** [comment]
It's reaches the title screen but it went black screen after press start.
![Screenshot_2022-12-05-11-51-35-420_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/205545953-c849610a-c403-4d1e-b4e2-e4c25fa1d5a6.jpg)

**Comment by benderscruffy** [comment]
open a new issue then
